lapis2/src/main/kotlin/org/genspectrum/lapis/controller/CompressionFilter.kt

```diff
@@ -222,6 +222,18 @@ class CompressingServletOutputStream(
         compressingStream.write(byte)
     }
 
+    override fun write(bytes: ByteArray) {
+        compressingStream.write(bytes)
+    }
+
+    override fun write(
+        bytes: ByteArray,
+        offset: Int,
+        length: Int,
+    ) {
+        compressingStream.write(bytes, offset, length)
+    }
```

Comment:
@chaoran-chen Those are the relevant changes. We didn't forward those methods to the compressing stream. Instead, the default implementation would call `fun write(byte: Int)` for every entry of the `ByteArray`. Apparently the compression streams can do a lot better than writing every byte individually.